blog/2024-05-16-blog-status-of-lbse-in-webkit.md

```diff
@@ -0,0 +1,149 @@
+
+Also in the same month, we attended the <a href="https://docs.webkit.org/Other/Contributor%20Meetings/ContributorMeeting2023.html">WebKit Contributors</a> meeting, where we presented a talk on the LBSE (slides are <a href="https://teams.pages.igalia.com/webkit/contributors-meeting-presentations/2023/igalia-slides/integrating-the-new-LBSE.html">here</a>) and the feedback on LBSE at the meeting was very positive. Giving the talk early on actually helped us since we needed to have a good design in place.
+
+## Supporting SVG resources
+
+The main focus in **October 2023** was introducing the SVG resource concept, as already outlined in the WebKit Contributors Meeting <a href="https://teams.pages.igalia.com/webkit/contributors-meeting-presentations/2023/igalia-slides/integrating-the-new-LBSE.html">talk</a>. Thus we started with <a href="https://commits.webkit.org/269522@main">adding the base SVG resource class</a>. The base class `RenderSVGResourceContainer` was kept as simple as possible with no support for resource invalidation or repainting logic. The main task of `RenderSVGResourceContainer` was to take care of registration so that the resource can be looked up by its clients.
+
+As the first SVG resource to implement we chose the SVG &lt;clip-path&gt;, so we <a href="https://commits.webkit.org/269635@main">landed RenderSVGResourceClipper</a>.
+
+To comply with the specification, the `RenderSVGResourceClipper` implementation produces 1-bit masks and uses a special rendering mode:
+
+- fill-opacity/stroke-opacity/opacity set to 1
+- masker/filter not applied when rendering the children
+- fill set to solid black and stroke set to none
+
+The initial implementation did not use caching of ImageBuffers for clipping, but relied on Porter-Duff DestinationIn/SourceOver compositing operations, to achieve the same effect, but faster. By integrating `RenderSVGResourceClipper` properly into `RenderLayer`, it aligned SVG clipping with HTML/CSS clipping.
+
+Finally, internally the implementation (like in legacy), prefers a pure clipping solution, but for more complicated clip-path's (for example when the clip-path involves text content) a fallback to a mask is done.
+
+## Resource invalidation handling
+
+After introducing the first SVG resource (`RenderSVGResourceClipper`) we noticed some issues with handling invalidations (for example adding to clip-path contents) for it. In the legacy engine, invalidations have been handled through layouting. This caused various problems, for one it could cause calling setNeedsLayout from within layout and the invalidation chain depended on the DOM order.
```

Comment:
should this be `.setNeedsLayout` maybe?  or even `.setNeedsLayout()`

Reply:
I now spell it out as being a method.